feat: Support explict modes for read fallbacks and caching (#104)
Layr-Labs/eigenda-proxy: README.md

An ephemeral memory store backend can be used for faster feedback testing when testing rollup integrations. To target this feature, use the CLI flags `--memstore.enabled`, `--memstore.expiration`.

### Storage Fallback

> An optional storage fallback CLI flag `--routing.fallback-targets` can be leveraged to ensure resiliency when **reading**. When enabled, a blob is persisted to a fallback target after being successfully dispersed using the keccak256 hash of the existing EigenDA commitment for the fallback target key. This ensure second order keys are succinct. In the event that blobs cannot be read from EigenDA, they will then be retrieved in linear order from the provided fallback targets. 

Reviewer: ```suggestion
An optional storage fallback CLI flag `--routing.fallback-targets` can be leveraged to ensure resiliency when **reading**. When enabled, a blob is persisted to a fallback target after being successfully dispersed. Fallback targets use the keccak256 hash of the existing EigenDA commitment as their key, for succinctness. In the event that blobs cannot be read from EigenDA, they will then be retrieved in linear order from the provided fallback targets. 
```

Reviewer: First reading was a bit confusing so rearranged. Not sure what you meant by "second order key". second order key sounds like the second key of a multi-map read to me, like dict[key1][key2]

Author: nice - appreciate the recommendation

Author: [265b7b9](https://github.com/Layr-Labs/eigenda-proxy/pull/104/commits/265b7b99a0f235eda7e3e3c4dfb9a2a89a883364)
